### PR TITLE
create default Cluster Agent dependencies

### PR DESCRIPTION
**Pull request by clamoriniere**
### What does this PR do?

Create default dependencies for the "cluster-agent" with
the v2 reconciler.

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Describe your test plan

Write there any instructions and details you may have to test your PR.

### CODE REVIEW

**Review by alai97 — APPROVED** [review]
Looks good, two comments!

File: controllers/datadogagent/dependencies/store.go [hunk 1]
```diff
@@ -83,6 +92,25 @@ func (ds *Store) AddOrUpdate(kind kubernetes.ObjectKind, obj client.Object) {
...
+	}
+	defaultAnnotations := object.GetDefaultAnnotations(ds.owner)
+	if obj.GetAnnotations() == nil {
+		obj.SetAnnotations(map[string]string{})
+	}
+	for key, val := range defaultAnnotations {
+		obj.GetAnnotations()[key] = val
+	}
+
+	ds.logger.Info("CEDTEST:", "kind", kind, "name", obj.GetName())
```
Comment (celenechang) [inline controllers/datadogagent/dependencies/store.go:108]: rm?

File: controllers/datadogagent/feature/enabledefault/feature.go [hunk 2]
```diff
@@ -19,13 +30,112 @@ func init() {
...
 
 func (f *defaultFeature) Configure(dda *v2alpha1.DatadogAgent) feature.RequiredComponents {
 	trueValue := true
+	f.namespace = dda.Namespace
+	f.owner = dda
+
+	f.clusterAgent.serviceAccountName = v2alpha1.GetClusterAgentServiceAccount(dda)
+	f.agent.serviceAccountName = v2alpha1.GetAgentServiceAccount(dda)
+	f.clusterCheckRunner.serviceAccountName = v2alpha1.GetClusterChecksRunnerServiceAccount(dda)
+
```
Comment (celenechang) [inline controllers/datadogagent/feature/enabledefault/feature.go:100]: should we check here if the component is disabled? `dda.Spec.Override[ClusterAgentComponentName].Disabled` or should we still configure this feature and make a decision in the `Override` function?

File: apis/datadoghq/common/envvar.go [hunk 3]
```diff
@@ -7,43 +7,105 @@ package common
...
+	DDAdmissionControllerInjectTags              = "DD_ADMISSION_CONTROLLER_INJECT_TAGS_ENABLED"
+	DDAdmissionControllerLocalServiceName        = "DD_ADMISSION_CONTROLLER_INJECT_CONFIG_LOCAL_SERVICE_NAME"
+	DDAdmissionControllerMutateUnlabelled        = "DD_ADMISSION_CONTROLLER_MUTATE_UNLABELLED"
+	DDAdmissionControllerServiceName             = "DD_ADMISSION_CONTROLLER_SERVICE_NAME"
+	DDAPIKey                                     = "DD_API_KEY"
+	DDAPMEnabled                                 = "DD_APM_ENABLED"
+	DDAppKey                                     = "DD_APP_KEY"
+	DDAuthTokenFilePath                          = "DD_AUTH_TOKEN_FILE_PATH"
+	DDClcRunnerEnabled                           = "DD_CLC_RUNNER_ENABLED"
+	DDCLCRunnerEnabled                           = "DD_CLC_RUNNER_ENABLED"
```
[the remark this answers is not in the document]
Comment (clamoriniere) [reply apis/datadoghq/common/envvar.go:23]: thanks merge issue...

**Review by celenechang — APPROVED** [review]

**Pull request merged**